Checkout/Sample/Source/3DS/WebViewController.swift

```diff
@@ -0,0 +1,90 @@
+//
+//  WebViewController.swift
+//
+//
+//  Created by Harry Brown on 01/03/2022.
+//
+
+import UIKit
+import WebKit
+import Checkout
+
+final class WebViewController: UIViewController {
+  struct ViewModel {
+    let onDismiss: ((_ webViewResult: Result<String, ThreeDSError>) -> Void)?
+    let webViewConfig: WebViewConfig
+
+    enum WebViewConfig {
+      case url(challengeURL: URL, successURL: URL, failureURL: URL)
+      case demo
+    }
+  }
+
+  var viewModel: ViewModel? {
```

Comment:
Not sure this is a genuine viewModel. I think it would be more intuitive if the user of the VC was kickstarting things like:
```
let vc = WebViewController()
let webConfig = WebViewConfig(.....) // WebViewConfig.demo
vc.load(config: webConfig) {
    // Completion handler
}
```

As such I would drop the ViewModel altogether and maybe use a typealias on `((_ webViewResult: Result<String, ThreeDSError>) -> Void)?` to have it more presentable.

Issue with injecting completion handlers via properties is how easy it is to create retain cycles.
- you don't use delegate which offers 1 point of failure to make all implementations safe from retain cycles
- you don't need `@escaping` which is enforced by compiler when it detects async code

This is similar with Reactive patterns but those usually come with reference collections to cleanup :).

Reply:
Also, small thing, but at call site assigning a property wouldn't be clear that will have consequences.

```
// Assigning
vc.model = FlashyModel()

// Clearly using a method and as such reasonably assuming things may happen - especially as its called start :)
vc.start(with: FlashyModel()
```

Reply:
Already exists, was just moving things. We can handle this in a follow up task